### PR TITLE
feat: add plugins detection

### PR DESCRIPTION
**Pull request by erezrokah**
Fixes https://github.com/netlify/framework-info/issues/109.

We could also have our own syntax for specifying conditions where a plugin should be used instead of extending `ajv`.

See deploy preview link https://deploy-preview-111--framework-info.netlify.app/

### Todo
- [x] Update the jsdoc comments
- [x] Update docs

### CODE REVIEW

File: src/frameworks/next.json [hunk 1]
```diff
@@ -14,5 +14,11 @@
     "command": "next build",
     "directory": "out"
   },
-  "env": {}
+  "env": {},
+  "plugins": [
+    {
+      "packageName": "@netlify/plugin-nextjs",
+      "schema": { "min_node_version": "10.13.0" }
```
Comment (ehmicky) [inline src/frameworks/next.json:21]: I like this declarative approach! :heart: 

Should `schema` be named according to what the property intent is (as opposed to its format, JSON schema), for example `condition`?

Comment (erezrokah) [reply src/frameworks/next.json]: Good point! 
Done in https://github.com/netlify/framework-info/pull/111/commits/4518b30cef2585279700f551c8a50ecfc041a62f

File: src/plugins.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+const MIN_NODE_VERSION_KEYWORD = {
+  keyword: 'minNodeVersion',
+  validate: (minNodeVersion, { nodeVersion }) =>
+    semver.valid(minNodeVersion) && semver.valid(nodeVersion) && semver.gte(nodeVersion, minNodeVersion),
+}
+
+const getPlugins = function (plugins, { nodeVersion }) {
+  const ajv = new Ajv({})
+  ajv.addKeyword(MIN_NODE_VERSION_KEYWORD)
```
Comment (ehmicky) [inline src/plugins.js:12]: Should those two lines be memoized?

`ajv` compiles JSON schemas under the hood, which is a little slow. It only does it the first time though (it is cached), so re-using the `ajv` instance between calls has a good performance advantage.

**Review by ehmicky — APPROVED** [review]
Great solution. Looks great!

I am guessing you are thinking of adding the other conditions for the Next.js plugin (`next.js` minimum version, etc.) as additional custom JSON schema keywords?

File: src/plugins.js (hunk 2 above)
Comment (erezrokah) [reply src/plugins.js:12]: Good point - I was probably too strict with not doing anything during `require` here

**Comment by erezrokah** [comment]
> I am guessing you are thinking of adding the other conditions for the Next.js plugin (`next.js` minimum version, etc.) as additional custom JSON schema keywords?

This PR puts us in a good place to implement additional conditions so we could do that - for `Next.js` version I would keep that logic inside the plugin.

File: src/frameworks/grunt.json [hunk 3]
```diff
@@ -11,5 +11,7 @@
     "command": "grunt build",
     "directory": "dist"
   },
-  "env": {}
+  "env": {},
+  "plugins": [],
+  "plugins": []
```
Comment (ehmicky) [inline src/frameworks/grunt.json:16]: @erezrokah I just noticed the duplicate there :arrow_up:

**Pull request merged**